demos/android/MASVS-CODE/MASTG-DEMO-0062/MASTG-DEMO-0062.md

```diff
@@ -0,0 +1,29 @@
+---
+platform: android
+title: Enforced Immediate Updates with Play Core API detected by semgrep
+id: MASTG-DEMO-0062
+code: [kotlin]
+test: MASTG-TEST-0290
+---
+
+### Sample
+
+The following code implements immediate in-app updates using the Google Play Core API by calling `startUpdateFlowForResult` with `AppUpdateOptions.newBuilder(1)`.
+
+### Steps
+
+Let's run @MASTG-TOOL-0110 rules against the sample code.
+
+{{ ../../../../rules/mastg-android-enforced-updating.yml }}
+
+{{ run.sh }}
+
+### Observation
+
+The output file shows usages of the Google Play Core API enforcing immediate update.
+
+{{ output.txt }}
+
+### Evaluation
+
+The test fails because the app forces users to update immediately and does not provide a fallback.
```

Comment:
The evaluation should be done based on the output. If the output cannot indicate a clear issue e.g., an insecure cryptographic algorithm being used, you may transfer the responsibility of evaluation to the tester. See DEMO-0007